### ISSUE TITLE
Set-Cookie header not found in WebResourceResponseReceived event e.response.headers

### ISSUE THREAD

**Issue by bittu** [MicrosoftEdge/WebView2Feedback#1802]
**Description**
Third party domain cookies are not getting saved by WebView2. It throws **The Set-Cookie is blocked by user preferences**  when i hover over the Set-Cookie header warning triangle.

So i am trying to read the headers using `WebResourceResponseReceived` event to save Cookie in a CookieContainer and trying to set the cookie in request on `WebResourceRequested` event.

But when i try to read the headers on `e.Response.Headers` there is no Set-Cookie available. Rest all the headers are present.

Is it the limitation of WebView2 for not able to read the Set-Cookie header or is it a bug.

***BTW the web app works in Edge Browser without any cookie issues***

**Version**
SDK: 1.0.902.49
Runtime: 94.0.992.31
Framework: WPF
OS: Win10

**Code snippet**
```cs
webView.CoreWebView2.AddWebResourceRequestedFilter("*", CoreWebView2WebResourceContext.All);
webView.CoreWebView2.WebResourceRequested += CoreWebView2_WebResourceRequested;
webView.CoreWebView2.WebResourceResponseReceived += CoreWebView2_WebResourceResponseReceived;

private void CoreWebView2_WebResourceRequested(object sender, CoreWebView2WebResourceRequestedEventArgs e)
{
    var reqUri = new Uri(e.Request.Uri);
    var cookie = getCookies(reqUri.Scheme + "://" + reqUri.Host); // returns cookie from CookieContainer
    Trace.WriteLine("Requested... Request... " + cookie);
    if (cookie != "" || cookie != null)
    {
        e.Request.Headers.SetHeader("Cookie", cookie);
    }
}

private void CoreWeView2_WebResourceResponseReceived(object sender, CoreWebView2WebResourceResponseReceivedEventArgs e)
{
    var headersStr = "";
    foreach (var header in e.Response.Headers)
    {
        headersStr += header.Key + ": " + header.Value + ", ";
    }
    foreach (var header in e.Request.Headers)
    {
        headersStr += header.Key + ": " + header.Value + ", ";
    }
    Trace.WriteLine("Received... Headers..." + headersStr); // this prints all headers except Set-Cookie
    if (e.Response.Headers.Contains("set-cookie") || e.Response.Headers.Contains("Set-Cookie")) // this is never true
    {
        var cookieHeader = e.Response.Headers.GetHeader("set-cookie");

        Trace.WriteLine("Received... Response... cookieHeader..." + cookieHeader);
        if (cookieHeader != "" || cookieHeader != null)
        {
          setCookie(cookieHeader);
        }
    }
}
```

[AB#37198783](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/37198783)

**Comment by champnic** [comment]
Hey @bittu - Would you be able to grab a net-log of your scenario?
https://textslashplain.com/2020/01/17/capture-network-logs-from-edge-and-chrome/

**Comment by Kindles7** [comment]
Hello. Same problem. The e.Request.Headers in the WebResourceResponseReceived event does not contain "set-cookie" header.
HAR file, from WebView2 DevTools. - [https://drive.google.com/file/d/126GJdMYt16M3w6inHbcIbLDOSKVnXzYS/view?usp=sharing](url)
![scr](https://user-images.githubusercontent.com/61284630/143774331-2d2aa355-eb2f-4c97-854d-c08fd8e14247.jpg)

**Comment by champnic** [comment]
I've added this bug to our backlog. Thanks for the log @Kindles7! If possible, could you also get a netlog? That will include more info and may include the decision for why a given header was or wasn't included in a network request.

**Comment by Kindles7** [comment]
@champnic ,here is NetLog, repeating actions for the missing "set-cookie" (in code).
[https://drive.google.com/file/d/1UvHTH8gkOBIJ2Xec_KLEj1ZVoKDW62hO/view?usp=sharing](url)

I noticed another problem. Tested on 3 computers (win10 and win11). At the first start, without a formed cache and a custom user agent, the call to Google is looped (err_too_many_redirects). If you do not specify the user agent, the Google page is loaded. And restarting with an already formed cache is supported by another user agent. This behavior was noticed only from Google, several other sites load normally with the same actions. I am attaching a test program and netlog.
[https://drive.google.com/file/d/1vf085dckRi0aZ6sC2VrQH0nC-Vo8EsQl/view?usp=sharing](url)
[https://drive.google.com/file/d/1swRIQG8s7QDTUoAK8oEcY_e8rPI9jA6m/view?usp=sharing](url)

**Comment by champnic** [comment]
Thanks for the netlog - I've confirmed there is a "set-cookie" header expected in the response that's not showing up in WebResourceResponseReceived. I'm following up with our networking folks to see if that's expected or if it's a bug.

**Comment by yildirimcagri** [comment]
Hi, this is a known issue that's currently being worked on. We expect a fix to be available within the next Edge Canary release. I'll keep you updated.

**Comment by yildirimcagri** [comment]
Hi again - this should be fixed with the latest release of [Edge Canary](https://www.microsoftedgeinsider.com/en-us/download/canary). Let me know if this works for you. You can change the WebView2 runtime to be used by setting the [WEBVIEW2_BROWSER_EXECUTABLE_FOLDER](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.1054.31#:~:text=BrowserExecutableFolder%20supports%20both%20relative%20%28to%20the%20application%27s%20executable%29,pass%20a%20null%20or%20empty%20string%20to%20browserExecutableFolder.) environment variable to the Edge Canary build directory (C:\Users\<user>\AppData\Local\Microsoft\Edge SxS\Application\<version_num>)

**Comment by Kindles7** [comment]
@yildirimcagri, thanks, everything worked.

**Comment by Thomas386** [comment]
Works fine for me too. Great! Is there a timeline when this fix is available in 'normal' a release?

**Comment by champnic** [comment]
The fix is in runtime versions 98.0.1091.0+, the stable version should ship around Feb 3, 2022.
